New module: Add module to collect facts for Amazon Redshift Cluster (cloud/amazon/redshift_facts)

j-carl opened the pull request:
Gather facts about Redshift Cluster(s).

##### SUMMARY
<!--- Describe the change, including rationale and design decisions -->
Gather facts about Redshift Cluster(s) either by:

- Get all cluster(s)
- Get cluster(s) by name (uses a regular expression)
- Get cluster(s) by tags

<!---
If you are fixing an existing issue, please include "Fixes #nnn" in your
commit message and your description; but you should still explain what
the change does.
-->

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->
 - New Module Pull Request

##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
`lib/ansible/modules/cloud/amazon/redshift_facts.py`

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.2.1.0
  config file =
  configured module search path = Default w/o overrides
```

j-carl committed:
New module redshift_facts

Gather facts about aws Redshift cluster(s).

j-carl committed:
Fix she-bang, PEP8 violations, and ansible metadata.

gundalow commented on lib/ansible/modules/cloud/amazon/redshift_facts.py:
```diff
@@ -0,0 +1,365 @@
+short_description: Gather facts about Redshift cluster(s)
+description:
+  - Gather facts about Redshift cluster(s)
+version_added: "2.3"
```
`"2.4"`

j-carl committed:
Bump version_added to "2.4"

s-hertel commented on lib/ansible/modules/cloud/amazon/redshift_facts.py:
```diff
@@ -0,0 +1,365 @@
+
+
+def match_tags(tags_to_match, cluster):
+    for key, value in tags_to_match.item():
```
I think this was a typo- should be tags_to_match.items():

j-carl replied:
Thx and I fixed it

s-hertel commented on lib/ansible/modules/cloud/amazon/redshift_facts.py:
```diff
@@ -0,0 +1,365 @@
+            tags=dict(type='dict')
+        )
+    )
+    module = AnsibleModule(argument_spec=argument_spec)
```
you could add supports_check_mode=True to AnsibleModule

j-carl replied:
I added the feature

s-hertel commented on lib/ansible/modules/cloud/amazon/redshift_facts.py:
```diff
@@ -0,0 +1,365 @@
+
+        matched_tags = True
+        if tags:
+            matched_tags = matched_tags(tags, cluster)
```
Another typo - should be match_tags(tags, cluster)

j-carl replied:
Thx

j-carl committed:
Fix some typos and add supporting check mode

j-carl committed:
Fix type of RETURN fields

Fix type of RETURN fields number_of_nodes and
automated_snapshot_retention_period from integer to int.

j-carl committed:
Add traceback information to fail_json()

s-hertel commented:
shipit

abadger commented:
Merged to devel for 2.4.0 release.
